### PR TITLE
chore: Move product category integration tests

### PR DESCRIPTION
**Pull request by olivermrbl**
Only missing tests are `DELETE`

### CODE REVIEW

**Comment by olivermrbl** [comment]
@sradevski, just saw you migrated a different product category test suite. I'll merge the two

File: integration-tests/http/__tests__/product/admin/product-category.spec.ts [hunk 1]
```diff
@@ -0,0 +1,956 @@
...
+import {
+  adminHeaders,
+  createAdminUser,
+} from "../../../../helpers/create-admin-user"
+
+jest.setTimeout(30000)
+
+medusaIntegrationTestRunner({
+  testSuite: ({ dbConnection, getContainer, api }) => {
+    let appContainer
```
Comment (sradevski) [inline integration-tests/http/__tests__/product/admin/product-category.spec.ts]: We can remove this as it's not used naywhere outside of `beforeEach`

File: integration-tests/http/__tests__/product/admin/product-category.spec.ts [hunk 2]
```diff
@@ -0,0 +1,956 @@
...
+      productModuleService = appContainer.resolve(
+        ModuleRegistrationName.PRODUCT
+      )
+
+      await createAdminUser(dbConnection, adminHeaders, appContainer)
+    })
+
+    describe("GET /admin/product-categories/:id", () => {
+      beforeEach(async () => {
+        productCategoryParent = await productModuleService.createCategory({
```
Comment (sradevski) [inline integration-tests/http/__tests__/product/admin/product-category.spec.ts]: Let's perform the creation through HTTP, so we are closer to what an end-user would do (and module changes won't break the tests if the HTTP layer remains constant)

**Comment by olivermrbl** [comment]
Btw. did you notice that `rank` is returned as string even though it is stored as `numeric` in the DB?

I found that strange. Thinking some serialization is done incorrectly but I can't dig it out

**Review by sradevski — APPROVED** [review]
Epic stuff, just the seeding can be done through HTTP, otherwise LGTM

File: integration-tests/http/__tests__/product/admin/product-category.spec.ts (hunk 2 above)
Comment (olivermrbl) [reply integration-tests/http/__tests__/product/admin/product-category.spec.ts]: Good push, will do

**Comment by sradevski** [comment]
@olivermrbl hey I missed your comments :) Yes please merge the two, I only moved the one that was written for v2. Weird about the rank, I didn't notice that, it is strange that it only happens for categories, probably something to do with the fact that we are using a custom repository layer (I see the module's tests are also expecting a string).

Update: https://github.com/mikro-orm/mikro-orm/issues/2295 it is on purpose in this case, we can serialize it to number

**Comment by olivermrbl** [comment]
@sradevski, thanks for pushing the fix. I updated the suite to only use the HTTP layer now and merged the two category suites. Should be good to merge.

**Review by sradevski — APPROVED** [review]
Re-approving, nice one 👍

**Pull request merged**